lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -103,6 +103,13 @@ sub needle_url {
 # Call to viewimg or viewaudio
 sub view {
     my ($self) = @_;
+
+    # Redirect users with the old preview link
+    if (!$self->req->is_xhr) {
+        my $target_url = $self->url_for('test', 'testid' => $self->param('testid'))."?".$self->param('moduleid')."/steps/".$self->param('stepid');
+        return $self->redirect_to($target_url);
+    }
```

Comment:
where is the test for this? you rewrote the test to rely on the new way only, but can you preserve the deprecated way in tests, too?